### PR TITLE
Preview html course files rather than download and embed

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12194
affects: Teacher
release note: Previewing html course files will resolve relative urls to other course files

### CODE REVIEW

File: rn/Teacher/src/modules/files/ViewFile.js [hunk 1]
```diff
@@ -284,6 +285,26 @@ export default class ViewFile extends Component<Props, State> {
...
+      case 'html':
+        // some people use files as a web server to serve html content and
+        // when they do they might include relative links to other files in the course
+        // which won't work with the download url. We must build a preview url that will
+        // redirect the webview to the server that serves file content in order for the
+        // relative urls to function properly
+        let { baseURL } = getSession()
+        let url = baseURL
+        if (this.props.context && this.props.contextID) {
+          url += `${this.props.context}/${this.props.contextID}/`
```
Comment (armstrongnate) [inline rn/Teacher/src/modules/files/ViewFile.js]: I believe file ids are unique for the account. I don't think we need to append the context. Do you know for sure?

Comment (shichongrui) [reply rn/Teacher/src/modules/files/ViewFile.js]: Yeah when I tried to do the preview in the browser with just `file/:fileID/preview` it failed.
https://ckruger.instructure.com/files/48901/preview
https://ckruger.instructure.com/courses/3313/files/48901/preview

Comment (shichongrui) [reply rn/Teacher/src/modules/files/ViewFile.js]: I do believe they are unique but the rails route/controller is probably set up to expect a context if it belongs to one or something like that.

Comment (armstrongnate) [reply rn/Teacher/src/modules/files/ViewFile.js]: Good to know! Thanks for clarifying.

**Comment by inst-danger** [comment]
#### Release Note: 
Previewing html course files will resolve relative urls to other course files

#### Affected Apps: Teacher

[MBL-12194](https://instructure.atlassian.net/browse/MBL-12194)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 94.14% | 94.14% | 0%
**Student** | 93.36% | 93.36% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by armstrongnate** [comment]
Here's a screen recording of what I'm seeing.

[ScreenRecording_04-05-2019 10-55-31.MP4.zip](https://github.com/instructure/canvas-ios/files/3048913/ScreenRecording_04-05-2019.10-55-31.MP4.zip)

**Pull request merged**